src/librados/librados.cc

```diff
@@ -624,6 +624,18 @@ uint32_t librados::NObjectIteratorImpl::seek(uint32_t pos)
   return r;
 }
 
+void librados::NObjectIteratorImpl::set_filter(const bufferlist &bl)
+{
+  assert(ctx);
+  if (ctx->nlc) {
+    ctx->nlc->filter = bl;
+  }
+
+  if (ctx->lc) {
+    ctx->lc->filter = bl;
```

Comment:
Are you planning to change the filter bufferlist encoding in the future?  If not, then you are ok.  Otherwise, there are backwards compatibility issues.

Even though you are only exposing the filter bufferlist to nobjects_begin() and related interface you still have to worry about supporting it in both CEPH_OSD_OP_PGNLS_FILTER and CEPH_OSD_OP_PGLS_FILTER.  This is because both of the over-the-wire protocols are used internally.